Stack/Opc.Ua.Core/Types/BuiltIn/NodeId.cs

```diff
@@ -10,6 +10,10 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
 
+// define to enable checks for a null NodeId modification
+// some tests are failing with this enabled, only turn on to catch issues
+// #define IMMUTABLENULLNODEID
```

Comment:
If some tests are failing aren't there issues with them ?

Reply:
that was the idea but it turned out some tests just failed because the new derived type was unknown. So the NodeId and ExpandedNodeId should really become immutable and sealed classes.